### PR TITLE
Feature/update to ink 4.0.0 beta

### PR DESCRIPTION
**Pull request by Artemka374**

### CODE REVIEW

File: contracts/src/security/reentrancy_guard/mod.rs [hunk 1]
```diff
@@ -65,7 +63,7 @@ where
 
     // We want to flush storage before execution of inner function,
     // because ink! doesn't do it by default and `status` will not be updated in child calls
-    push_spread_root(instance.data(), &Default::default());
+    // todo: ink::env::set_contract_storage(Default::default(), &instance.data());
```
Comment (coreggon11) [inline contracts/src/security/reentrancy_guard/mod.rs:66]: todo?

File: docs/docs/smart-contracts/example/impls.md [hunk 2]
```diff
@@ -320,7 +320,7 @@ impl lending::Internal for LendingContract {
     fn _instantiate_shares_contract(&self, contract_name: &str, contract_symbol: &str) -> AccountId {
         let code_hash = self.lending.shares_contract_code_hash;
         let (hash, _) =
-            ink_env::random::<ink_env::DefaultEnvironment>(contract_name.as_bytes()).expect("Failed to get salt");
+            ink::env::random::<ink::env::DefaultEnvironment>(contract_name.as_bytes()).expect("Failed to get salt");
```
Comment (coreggon11) [inline docs/docs/smart-contracts/example/impls.md:323]: changed this

File: examples/alternatives/diamond/rust/lib.rs [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
 mod ext;
 
 #[cfg(not(feature = "std"))]
-use ink_primitives::{
-    Key,
-    KeyPtr,
-};
-#[cfg(not(feature = "std"))]
-use ink_storage::traits::SpreadLayout;
+use ink::primitives::Key;
```
Comment (coreggon11) [inline examples/alternatives/diamond/rust/lib.rs:10]: format pls

File: examples/alternatives/diamond/rust/lib.rs [hunk 4]
```diff
@@ -19,34 +17,37 @@ use openbrush::traits::{
...
+        let (_, facet_cut) = ink::env::decode_input::<([u8; 4], FacetCut)>().unwrap();
 
         // Support of diamond
-        let mut root_key: KeyPtr = KeyPtr::from(ROOT_KEY);
-        let mut storage = <diamond::Data as SpreadLayout>::pull_spread(&mut root_key);
+        let mut storage = ink::env::get_contract_storage::<Key, diamond::Data>(&ROOT_KEY)
+            .unwrap()
+            .unwrap();
         storage._diamond_cut_facet(&facet_cut).expect("Init diamond cut");
-        storage.flush();
```
Comment (coreggon11) [inline examples/alternatives/diamond/rust/lib.rs:35]: why no flush

File: tests/diamond_loupe.rs [hunk 5]
```diff
@@ -252,6 +252,7 @@ mod diamond {
         assert_eq!(diamond.facets()[0].selectors[2], [4u8; 4]);
     }
 
+    #[ignore]
```
Comment (coreggon11) [inline tests/diamond_loupe.rs:255]: why ignore

**Review by coreggon11 — COMMENTED** [review]
pls review and update docs (constructors and toml)

File: docs/docs/smart-contracts/example/contract.md [hunk 6]
```diff
@@ -48,12 +41,12 @@ crate-type = [
 [features]
 default = ["std"]
 std = [
-    "ink_primitives/std",
-    "ink_metadata",
-    "ink_metadata/std",
-    "ink_env/std",
-    "ink_storage/std",
-    "ink_lang/std",
+    "ink::primitives/std",
```
[the remark this answers is not in the document]
Comment (Artemka374) [reply docs/docs/smart-contracts/example/contract.md:44]: ctrl + shift + R 😄

File: .github/workflows/docker_image.yml [hunk 7]
```diff
@@ -9,14 +9,14 @@ jobs:
       - uses: actions/checkout@v2
       - id: files
         uses: jitterbit/get-changed-files@v1
-      - name: Build & Push Docker Image
+      - name: Build & Push DockerF Image
```
Comment (coreggon11) [inline .github/workflows/docker_image.yml:12]: change

**Comment by coreggon11** [comment]
@xgreenx CI is passing, could you review please? 😄

**Review by coreggon11 — DISMISSED** [review]

**Review by coreggon11 — APPROVED** [review]

**Pull request merged**